targets/TARGET_STM/TARGET_STM32H7/i2c_device.c

```diff
@@ -76,6 +76,30 @@ uint32_t i2c_get_pclk(I2CName i2c)
                 break;
         }
     }
+#endif
+#if defined I2C5_BASE
+    else if (i2c == I2C_5) {
+        clocksource = __HAL_RCC_GET_I2C5_SOURCE();
+        switch (clocksource) {
+            case RCC_I2C5CLKSOURCE_D2PCLK1:
+                pclk = HAL_RCCEx_GetD2PCLK1Freq();
```

Comment:
@jeromecoutant `HAL_RCCEx_GetD2PCLK1Freq()` does not seem to exist anywhere. 

Similar function exist, but not for D2PCLK1:
https://github.com/ARMmbed/mbed-os/blob/c6d814e68c9b298f2690a9ad5f08299072af4a71/targets/TARGET_STM/TARGET_STM32H7/STM32Cube_FW/STM32H7xx_HAL_Driver/stm32h7xx_hal_rcc_ex.h#L3899

Reply:
Good catch.
See #15229